meetings/2024-12-10/minutes.md

```diff
@@ -0,0 +1,62 @@
+# OQS Technical Steering – 2024-12-10 – minutes
+
+## Attendees
+
+* Douglas Stebila (U Waterloo)
+* Spencer Wilson (U Waterloo)
+* Michael Baentsch (Intependent)
+* Brian Jarvis (Amazon)
+* Christian Paquin (Microsoft)
+* Norman Ashley (Cisco)
+* Alex Bozarth (IBM)
+* Basil Hess (IBM)
+
+## Agenda
+
+1. Chair's introduction
+
+- Introduction by Douglas.
+
+2. Approve agenda
+
+- No changes requested.
+
+3. Appoint minute-taker
+
+- Basil Hess
+
+4. Review action items from previous meeting
+
+- [Security response team](https://github.com/open-quantum-safe/tsc/issues/60)
+
+- Setup GitHub/email alias: Pending.
+- Dry run pending security response policy. Spencer and Douglas to discuss.
+- Spencer: Awaiting wet run of HQC incident; will draft process document (PR forthcoming).
+- Douglas: Solicit feedback, expecially from people with prior experience.
+- Michael: Suggests documenting guidelines on how to interact with upstream projects. Responsiveness from upstreams is a concern; to identify key contacts at upstreams.
+
+5. Reports (PQCA TAC, PQ Code Package) - Spencer
+
+- TAC discussion around supporting context strings in ML-KEM. Google's Tink library doesn't expose context string features, alghough Tink aims at providing higher-level API compared to liboqs.
+- PQCP mlkem-native version 1.0-alpha has been released, with Pravek and Basil working on its integration with OQS.
+
+
+6. [SLH-DSA and which upstream code bases to rely on](https://github.com/open-quantum-safe/liboqs/issues/1894)
+
+- Discussion on whether to rely on OpenSSL’s upcoming SLH-DSA implementation or explore other upstream sources.
+- Considerations include performance, diversity, formal verification, and composite use cases.
+- Douglas will reach out to SPHINCS+ authors for input.
+- Michael suggests exploring the option of dropping support for SLH-DSA.
```

Comment:
The complete context here was Christian's comment/question whether it is worthwhile continuing work on OQS at all given that not only SLH-DSA, but all standardized PQC algorithms are now being added to OpenSSL -- which OQS already now relies on/falls back to by default and for "beyond research use" cases.

I think this thought also needs to be captured in the minutes as a) also concerns me (and I think OQS users) and b)  requires serious thought going forward: Sample questions: What use(r)s should OQS focus on/cater for? Does OQS want to compete with, complement or use OpenSSL? In general or differently per algorithm? In case of resource (people) contention, what to prioritize? The same question is valid (and already being answered) for oqs-boringssl: There the upstream code is being prioritized (unless I misunderstand the intentions by @pi-314159 -- maybe you want to chime in?).

Also, I think @dstebila rightfully mentioned that it'd be pretty pointless for oqs-provider to keep supporting algorithms that are already implemented by OpenSSL's default provider (if `liboqs` already gets its code from OpenSSL (default provider), so that also should be considered.

Either way, I think this warrants an action item (discussion in the near future how to prioritize to not waste development effort).

Reply:
Based on my understanding and usage (or @SWilson4's usage), we test different algorithms, with a particular focus on experimentation rather than production use.

If OQS' goal is to create a testbed or playground for exploring various cryptographic algorithms, then it remains meaningful and should continue to exist—perhaps even after NIST's standardization process. This would allow us to experiment with algorithms that are not standard, but have not been proven to have any security vulnerabilities.

However, if OQS' goal is to support production use, then we should consider dissolving this organization, as other projects like BoringSSL and OpenSSL have already implemented the relevant standards.

Reply:
Thanks, @baentsch, for pointing this out. I re-listened to that part of the recording and realized some context was indeed missing. I've now updated the minutes to include more detailed context from the conversation.